### PR TITLE
add fix for mdmonitor.service failure

### PR DESCRIPTION
**Pull request by zimbatm**
See https://github.com/NixOS/nixpkgs/issues/72394#issuecomment-549110501

### CODE REVIEW

File: common/mdmonitor_fix.nix [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  # See https://github.com/NixOS/nixpkgs/issues/72394#issuecomment-549110501
+  environment.etc."mdadm.conf".text = ''
+    MAILADDR root
+  '';
+}
```
Comment (Mic92) [inline common/mdmonitor_fix.nix:6]:
```suggestion
{ lib, config, ... }: {
  # See https://github.com/NixOS/nixpkgs/issues/72394#issuecomment-549110501
  environment.etc = lib.mkIf (config.systemd.services.mdmonitor.enable) {
    "mdadm.conf".text = ''
      MAILADDR root
    '';
  };
}
```

File: common/default.nix [hunk 2]
```diff
@@ -4,6 +4,7 @@
 {
   imports = [
     ./flake.nix
+    ./mdmonitor_fix.nix
```
Comment (Mic92) [inline common/default.nix:7]:
```suggestion
    ./mdmonitor-fix.nix
```

**Comment by zimbatm** [comment]
bors merge

File: common/mdmonitor-fix.nix [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{ lib, config, ... }: {
+  # See https://github.com/NixOS/nixpkgs/issues/72394#issuecomment-549110501
+  config = lib.mkIf config.systemd.services.mdmonitor.enable {
```
Comment (Mic92) [inline common/mdmonitor-fix.nix:3]: Mhm. looks like mdadm.conf is depending on this.

File: common/mdmonitor-fix.nix [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    environment.etc."mdadm.conf".text = lib.mkDefault ''
+      MAILADDR root
+    '';
```
Comment (Mic92) [inline common/mdmonitor-fix.nix:6]:
```suggestion
    boot.initrd.services.swraid.mdadmConf = ''
      MAILADDR root
    '';
```

**Comment by zimbatm** [comment]
bors merge

**Pull request merged**